### ISSUE TITLE
Bug: Json output mode incorrectly serializes arrays

### ISSUE THREAD

**Issue by elviejo79** [kuzudb/kuzu#4634]
### Kùzu version

v0.7.0

### What operating system are you using?

ubuntu 20.04

### What happened?

When executing the kuzu REPL

```
$ kuzu
Opening the database under in-memory mode.
Enter ":help" for usage hints.
kuzu> :mode json
mode set as json
kuzu> RETURN [1,2,3] as should_be_an_array_not_a_string;
[{"should_be_an_array_not_a_string":"[1,2,3]"}]
```
# What I exepected to happen?
the serialization of an array is without quotes
```
[{"should_be_an_array_not_a_string":[1,2,3]}]
```
Observe that it should be `[1,2,3]` not `"[1,2,3]"`

**Comment by elviejo79** [comment]
on the other side the json extension does the 'right thing':
``` bash
$ kuzu 
Opening the database under in-memory mode.
Enter ":help" for usage hints.
kuzu> :mode json
mode set as json
kuzu> INSTALL json;
[{"result":"Extension: json has been installed."}]
(1 tuple)
(1 column)
Time: 0.06ms (compiling), 1049.62ms (executing)
kuzu> LOAD EXTENSION json;
[{"result":"Extension: json has been loaded."}]
(1 tuple)
(1 column)
Time: 0.04ms (compiling), 1.13ms (executing)
kuzu> COPY (RETURN [1,2,3] as should_be_an_array) TO 'output.json';
kuzu> :quit
$ cat output.json                                                          
[
{"should_be_an_array":[1,2,3]}
]
```

**Comment by elviejo79** [comment]
refs #4053 .
What do you think @MSebanc ?

**Comment by andyfengHKU** [comment]
Hi @elviejo79, I think what you suggest make sense. @MSebanc is a bit busy with a couple of things these days. @acquamarin and @SterlingT3485 will take a look and fix this soon.

**Comment by acquamarin** [comment]
Hi @elviejo79 
The bug has been solved #4643
